src/utils/style-utils.js

```diff
@@ -1,137 +1,72 @@
-import isImmutableMap from './is-immutable-map';
-import diffStyles from './diff-styles';
 
-export function getInteractiveLayerIds(mapStyle) {
-  let interactiveLayerIds = null;
-
-  if (isImmutableMap(mapStyle) && mapStyle.has('layers')) {
-    interactiveLayerIds = mapStyle.get('layers')
-      .filter(l => l.get('interactive'))
-      .map(l => l.get('id'))
-      .toJS();
-  } else if (Array.isArray(mapStyle.layers)) {
-    interactiveLayerIds = mapStyle.layers.filter(l => l.interactive)
-      .map(l => l.id);
-  }
-
-  return interactiveLayerIds;
-}
-
-// Individually update the maps source and layers that have changed if all
-// other style props haven't changed. This prevents flicking of the map when
-// styles only change sources or layers.
-/* eslint-disable max-statements, complexity */
-export function setDiffStyle(prevStyle, nextStyle, map) {
-  const prevKeysMap = prevStyle && styleKeysMap(prevStyle) || {};
-  const nextKeysMap = styleKeysMap(nextStyle);
-  function styleKeysMap(style) {
-    return style.map(() => true).delete('layers').delete('sources').toJS();
+export function getInteractiveLayerIds(style) {
```

Comment:
Maybe a comment header with the thinking here, from a passing read of another PR I believe you wanted to deprecate this in the near future/